common/lib/xmodule/xmodule/modulestore/mongo/draft.py

```diff
@@ -624,6 +624,9 @@ def has_changes(self, xblock):
             return True
         # if this block doesn't have changes, then check its children
         elif xblock.has_children:
+            # fix a bug where dangling pointers should imply a change
+            if len(xblock.children) > len(xblock.get_children()):
```

Comment:
I'll document my understanding of the above for Don's approval:

First, `xblock.has_children` means "Is this XBlock allowed to have children?"
`xblock.children` is the actual list of children, including any non-existent children.
`xblock.get_children()` returns a list of valid children, which might be less than the actual list, since it doesn't return the invalid children.
So - yes - should be a subset.